From 487a9ea05011e94edfa130d1e4e9a92efa9f5ddb Mon Sep 17 00:00:00 2001
Subject: [PATCH 12/13] The legacy interface makes DataSize "unsigned long"
 not uint64_t.

This probably hasn't ever been tested on i386 before, but Steve McIntyre
did and it didn't work, because kernel says "unsigned long" for
DataSize, and we have uint64_t.

This should fix issue #11 .

Signed-off-by: Curt Brune <curt@cumulusnetworks.com>

diff --git a/src/vars.c b/src/vars.c
index 97487f0..ad344a0 100644
--- a/src/vars.c
+++ b/src/vars.c
@@ -34,7 +34,7 @@
 typedef struct efi_kernel_variable_t {
 	uint16_t	VariableName[1024/sizeof(uint16_t)];
 	efi_guid_t	VendorGuid;
-	uint64_t	DataSize;
+	unsigned long	DataSize;
 	uint8_t		Data[1024];
 	efi_status_t	Status;
 	uint32_t	Attributes;
